### PR TITLE
Improve error message for missing return type annotations of `@step` functions

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
I added an error message for when the return type annotation of an `@step` function is missing.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/steps/base_step.py [hunk 1]
```diff
@@ -164,6 +164,13 @@ def __new__(
                 cls.INPUT_SIGNATURE.update({arg: arg_type})
 
         # Parse the returns of the step function
+        if "return" not in step_function_signature.annotations:
+            raise StepInterfaceError(
+                f"Missing return type annotation when "
+                f"trying to create step '{name}'. Please make sure to "
+                f"include type annotations for all your step inputs "
+                f"and outputs."
```
Comment (htahir1) [inline src/zenml/steps/base_step.py:172]:
```suggestion
                f"and outputs. If your step returns nothing, please annotate it with `-> None`."
```

File: src/zenml/steps/step_decorator.py [hunk 2]
```diff
@@ -86,6 +86,9 @@ def step(
     Returns:
         the inner decorator which creates the step class based on the
         ZenML BaseStep
+
+    Raises:
+        StepInterfaceError: raises an error if _func is not type annotated.
```
Comment (htahir1) [inline src/zenml/steps/step_decorator.py:91]:
```suggestion        
```

File: src/zenml/steps/step_decorator.py [hunk 3]
```diff
@@ -97,6 +100,9 @@ def inner_decorator(func: F) -> Type[BaseStep]:
 
         Returns:
             The class of a newly generated ZenML Step.
+
+        Raises:
+            StepInterfaceError: raises an error if _func is not type annotated.
```
Comment (htahir1) [inline src/zenml/steps/step_decorator.py:105]:
```suggestion
```

File: src/zenml/steps/step_decorator.py (hunk 2 above)
Comment (htahir1) [inline src/zenml/steps/step_decorator.py:91]:
```suggestion
```

**Review by htahir1 — CHANGES_REQUESTED** [review]
Great error message! Just small nits

**Review by schustmi — APPROVED** [review]

**Review by htahir1 — APPROVED** [review]

**Pull request merged**